ivy/functional/ivy/data_type.py

```diff
@@ -2102,6 +2102,9 @@ def promote_types(
     ret
         The type that both input types promote to
     """
+    # in case either is of none type
+    if not (type1 and type2):
+        return type1 if type1 else type2
```

Comment:
wouldn't `return type1 if type1 else type2` on this line be the same as `return` alone?

Reply:
didn't get you, the logic here is to return either of type1 or type2 if either of them are None.

Reply:
ah yeah, my bad sorry!